htlcswitch/link.go

```diff
@@ -1475,6 +1491,40 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 		}
 
 	case *lnwire.CommitSig:
+		// Since we may have learned new preimages for the first time,
+		// we'll add them to our preimage cache. By doing this, we
+		// ensure any contested contracts watched by any on-chain
+		// arbitrators can now sweep this HTLC on-chain. We delay
+		// committing the preimages until just before accepting the new
+		// remote commitment, as afterwards the peer won't resend the
+		// Settle messages on the next channel reestablishment. Doing so
+		// allows us to more effectively batch this operation, instead
+		// of doing a single write per preimage.
+		if len(l.uncommittedPreimages) > 0 {
+			err := l.cfg.PreimageCache.AddPreimages(
```

Comment:
Does the `AddPreimages` method handle zero-length slices? If it did, we can avoid some indentation here...

Reply:
good suggestion, added a check in `AddPreimages` to exit early for zero length slices